clusterloader2/testing/load/experimental/statefulset.yaml

```diff
@@ -0,0 +1,55 @@
+{{$EnablePVs := DefaultParam .ENABLE_PVS false}}
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{.Name}}
+  labels:
+    group: statefulset
+spec:
+  selector:
+    matchLabels:
+      name: {{.Name}}
+  serviceName: {{.Name}}
+  replicas: {{RandIntRange .ReplicasMin .ReplicasMax}}
+  template:
+    metadata:
+      labels:
+        group: statefulset
+        name: {{.Name}}
+    spec:
+      terminationGracePeriodSeconds: 1
+      containers:
+      - name: {{.Name}}
+        image: k8s.gcr.io/pause:3.1
+        ports:
+          - containerPort: 80
+            name: web
+        {{if $EnablePVs}}
+        volumeMounts:
+          - name: pv
+            mountPath: /var/pv
+        {{end}}
+        resources:
+          requests:
+            cpu: 10m
+            memory: "10M"
+  {{if $EnablePVs}}
+  # TODO(#704): We should have better control over deleting PVs.
+  # While this StatefulSet's feature is convenient for creating PVs, there is no easy way to delete
+  # them later, as deleting StatefulSet doesn't delete PVs. They are going to be eventually deleted
+  # once namespace is deleted, but ideally we should have a better (more controlled) way of doing
+  # that, so we can delete them during test, e.g.  #             after StatefulSets are deleted, and
+  # measure api call latency for that. There are two options here:
+  # 1. Create a new measurement that will delete PVs created by StatefulSets
+  # 2. Create PVs manually, attach them here (but not via volumeClaimTemplates) and delete them
+  #    manually later
```

Comment:
I actually think that there is a third option which is faster and simpler to do (and I would vote for proceeding this path at least for now).
PVCs connected to StatefulSet have naming schema: "<pvc-name-from-template>-statefulset-name>-<index>".

So we can actually create a template file for PVC and we will simply add a step to delete PVCs that would look like:
```
    namespaceRange:
      min: 1
      max: {{$namespaces}}
    replicasPerNamespace: 0
    tuningSet: RandomizedSaturationTimeLimited
    objectBundle:
      - basename: pvname-ssname
        objectTemplatePath: pvc.yaml
```

If at some point we decide for more pvc-per-pod or more statefulset-per-namespace, we can opaque it in `range` (ugly but works - I don't we need it immediately).


The only remaining problem we have is that we enumrate from 1 to N, and StatefulSet from 0 to N-1.
But I think we can change that universally for CL, to actually use 0..N-1 scheme (I think that's reasonable on itself).

Reply:
Done, but from what I managed to get from code it looks like we're enumerating from 0 in CL2 already. Please correct me if I'm wrong

Reply:
I think it used to be 1..N, but apparently it's not:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/test/simple_test_executor.go#L199

Reply:
As discussed f2f, unfortunately this wouldn't work, as CL2 is not able to delete objects that it didn't create. Long story short it keeps it own state of all objects that were created/modified. When we try to delete the PVCs, CL2 doesn't know about them and assumes that none of them exist and doesn't do anything when deleting. 

Leaving the TODO as it was, will tackle it in a separate PR.